src/main/java/com/google/songgame/servlets/RoomServlet.java

```diff
@@ -1,25 +1,62 @@
 package com.google.songgame.servlets;
 
 import com.google.gson.Gson;
+import com.google.gson.JsonParser;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
 import java.io.IOException;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.stream.Collectors;
+import java.util.Collections;
 
 import org.apache.hc.core5.http.ParseException;
 import com.google.gson.Gson; 
-import java.util.ArrayList;
+import java.util.*;
 
 @WebServlet("/room")
 public final class RoomServlet extends HttpServlet {
 
+  private Gson gson;
+
+  @Override
+  public void init() {
+    gson = new Gson();
+  }
+
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    // Get user name
-    String name = request.getParameter("user-name");
-    response.setContentType("text/html;");
-    response.getWriter().println("Name: " + name);
+    Map<String, String> userJson = readJSONFromRequest(request);
+
+    // Save player username and userId to datastore.
+    Entity userEntity = new Entity("User");
+    userEntity.setProperty("username", getValuesList(userJson).get(0));
+    userEntity.setProperty("userId", getValuesList(userJson).get(1));
+    
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(userEntity);
+
+    response.sendRedirect("/lobby.html");
+  }
+
+  private Map<String, String> readJSONFromRequest(HttpServletRequest request) throws IOException {
+    String requestJSONString = request.getReader().lines().collect(Collectors.joining());
+    Map jsonData = gson.fromJson(requestJSONString, Map.class);
```

Comment:
I think it's better to parse this as a Map<String, String> like we recommended Salil to do in his recent change.

You can take a look at this line for an example of what to do:
https://github.com/googleinterns/step105-2020/blob/master/src/main/java/com/google/songgame/servlets/ChatServlet.java#L52

Let us know if you have any questions about this

Ideally we would move readJSONFromRequest() into a common class that could be shared across the different servlets, but we can make it a stretch goal to do this